app/views/helpers/feed/update.phtml

```diff
@@ -48,7 +48,7 @@
 					<a class="btn" target="_blank" href="<?php echo $this->feed->url(); ?>"><?php echo _i('link'); ?></a>
 				</div>
 
-				<a class="btn" target="_blank" href="http://validator.w3.org/feed/check.cgi?url=<?php echo $this->feed->url(); ?>"><?php echo _t('sub.feed.validator'); ?></a>
+				<a class="btn" target="_blank" href="http://validator.w3.org/feed/check.cgi?url=<?php echo urlencode(str_replace('&amp;', '&', $this->feed->url())); ?>"><?php echo _t('sub.feed.validator'); ?></a>
```

Comment:
Interestingly, the validator itself opts for `+` instead of `%20`. But yes, I didn't realize `urlencode` didn't abide by the URI RFC.